src/group_theory/subgroup/basic.lean

```diff
@@ -830,6 +830,39 @@ begin
   { simp only [true_and, coe_to_submonoid, union_subset_iff, subset_closure, inv_subset_closure] }
 end
 
+@[to_additive] lemma closure_induction_left {p : G → Prop} {x : G}
+  (h : x ∈ closure k) (H1 : p 1) (Hmul : ∀ (x ∈ k) y, p y → p (x * y))
+  (Hinv : ∀ (x ∈ k) y, p y → p (x⁻¹ * y)) : p x :=
+begin
+  have := le_of_eq (closure_to_submonoid k),
+  obtain ⟨l, hl, rfl⟩ := submonoid.exists_list_of_mem_closure (this h),
```

Comment:
Can you extra a `submonoid.closure_induction_left` principle here, and use that to golf this further?